tests/e2e/restart_test.go

```diff
@@ -84,7 +84,7 @@ func (s *RestartTestData) TestVolumeDetached(c *C) {
 	c.Assert(err, IsNil, Commentf(out))
 
 	// 4. Verify detached status. Volume should be detached (within the timeout)
-	status = verification.VerifyDetachedStatus(s.volumeName, s.config.DockerHosts[1], s.config.EsxHost)
+	status = verification.PollDetachedStatus(s.volumeName, s.config.DockerHosts[1], s.config.EsxHost)
```

Comment:
@govint 
> A new function is there in utils/docker/containers.go:GetVolumeStatus() which will return a map of the volume properties including the volume state. Use that to get the instantaneous state of a volume. Why keep adding more functions.

I agree adding more redundant utils is useless. But here we need to verify detached status from both docker host side and esx side. That's exactly VerifyDetachedStatus is doing.

> A single API that takes a waitTillVolumeDetached parameter

Existing VerifyDetachedStatus always waits till volume gets detached (within timeout). Adding an extra parameter to decide whether to wait or not would change the func signature and still induce test code change (but we will achieve the goal of single util to do single work). And the if/else part would always hit through this util.

Since most of the cases we need to instantly determine the detached status and few of them need to poll (restart and failover scenarios), I introduced PollDetachedStatus.
We can do also it through the one API way and have extra parameter and if block inside, but I think two separate utils is cleaner. Let me know what you think about this.